### PR TITLE
[DOCS] Reformat unique token filter docs

### PR DESCRIPTION
**Pull request by jrodewig**
### Changes
* Updates the description
* Adds note re: `remove_duplicates` filter
* Adds analyze, custom analyzer, and custom filter snippets
* Adds parameter documentation

### CODE REVIEW

File: docs/reference/analysis/tokenfilters/unique-tokenfilter.asciidoc [hunk 1]
```diff
@@ -4,7 +4,147 @@
...
+filter to change `the lazy lazy dog` to `the lazy dog`.
+
+If the `only_on_same_position` parameter is set to `true`, the `unique` filter
+removes only duplicate tokens _in the same position_.
+
+[NOTE]
+====
+When `only_on_same_position` is `true`, the `unique` filter works the same as
+<<analysis-remove-duplicates-tokenfilter,`remove_duplicates`>> filter.
+====
```
Comment (jrodewig) [inline docs/reference/analysis/tokenfilters/unique-tokenfilter.asciidoc]: @romseygeek Do you know if there's any reason for users to prefer one of these filters over another? If so, I can note that here.

**Comment by jrodewig** [comment]
@romseygeek Any feedback on this one? Thanks!

File: docs/reference/analysis/tokenfilters/unique-tokenfilter.asciidoc (hunk 1 above)
Comment (romseygeek) [reply docs/reference/analysis/tokenfilters/unique-tokenfilter.asciidoc]: Sorry, missed this! `unique` has the ability to work over an entire token stream, while `remove_duplicates` looks only at consecutive tokens, but other than that they are identical.

**Review by romseygeek — APPROVED** [review]
LGTM

**Comment by jrodewig** [comment]
Thanks @romseygeek!

**Comment by jrodewig** [comment]
### Backport commits
master c99a0e9
7.x 70e4ae3381c
7.6 aec54a20076
7.5 aa116a4592d

**Pull request merged**